### ISSUE TITLE
Osvětlení multikoptéry

### ISSUE THREAD

**Issue by taury55** [ThunderFly-aerospace/TF-M8X#6]
Multikoptéře chybí aktuálně jakákoliv signalizace stavu v jakém se nachází a při horších světelných podmínkách je náročné vidět aktuální její aktuální polohu. Pro by se na konec každého ramena měla předat RGB dioda pro signalizaci stavu a vizualizaci polohy v prostoru. Také by mělo být přidáno přistávací osvětlení pro osvícení terénu pod multikoptérou.

**Comment by roman-dvorak** [comment]
Tohle issue by mohlo být vyřešeno tak, že se navrhne I2C-PWM převodník, se kterým bude autopilot komunikovat přímo po I2C. Předpokládám, že na výstupu by převodník měl mít nějaké výkonové tranzistory pro spínání LED diod. 

Veškerý SW by měl být ve FW autopilota. Ovladač by zřejmě měl mít několik režimů. Pro letadla a multikoptéry. Na základě toho by bylo zvolen způsob blikání. Podle způsobu blikání by mělo být možné poznat nějaké základní stavy dronu.

Jeden z možných režimů pozičních LEDek (pro letadla) je, že LEDky budou stále svítit menším výkonem a periodicky, na krátkou dobu, probliknou vyšším výkonem.

**Comment by kaklik** [comment]
Další požadavky na konstrukci jsou: 

-  Možnost připojit výkonovou LED nebo LED pásek, ideálně přes konektory, aby bylo možné odpojit vodiče a protáhnout je skrz konstrukci
-  Napájení modulu by mělo být výkonové, z toho důvodu bude modul potřebovat ještě XT konektor, nebo autopilotí napájecí konektor
-  V případě připojení na I2C musí modul mít možnost volby více adres

##### Současné řešení

PX4 má interní driver pro RGB led budič [NCP5623C](https://www.onsemi.com/pub/Collateral/NCP5623-D.PDF), který ale má pouze jednu I2C adresu, není také konstruovaný pro spínání výkonových led o celkovém výkonu třeba 10W.

**Comment by kaklik** [comment]
##### Nalezené integrované obvody

- [LT3964](https://www.analog.com/media/en/technical-documentation/data-sheets/LT3964.pdf)
- [PCA9635]( https://github.com/MLAB-project/Modules/blob/master/human_interface/LED7SEG01A/doc/pdf/PCA9635.pdf)
- [TPS92515AHV](http://www.ti.com/lit/ds/symlink/tps92515ahv-q1.pdf) - PWM budič LED
- [tps92518hv](http://www.ti.com/lit/ds/symlink/tps92518hv-q1.pdf) - SPI nebo PWM LED budič 

U I²C budičů je možné, že nestací šířka pásma sběrnice pro obsluhu všech LED.  Problematickým módem může být zábleskové blikání, které by mělo mít délku maximálně desítky milisekund a musí být velmi intenzivní. 

Pokud I2C nebude stačit, tak je možné použít SPI a u autopilotů bez SPI toto rozhraní vytvořit I2C-SPI převodníkem za cenu omezení funkcí. 

##### Nalezené světelné zdroje
 
- https://www.svetzarovek.eu/p-80055/k-1408/cob-led-pasek-12v-11w/

**Comment by roman-dvorak** [comment]
> U I²C budičů je možné, že nestací šířka pásma sběrnice pro obsluhu všech LED. Problematickým módem může být zábleskové blikání, které by mělo mít délku maximálně desítky milisekund a musí být velmi intenzivní.

Některé I2C PWM převodníky mají funkci, že umí ovladat několik výstupů (skupinu) zároveň. Což docela řeší tento problém. 

Jinak bych docela vážně uvažoval o možnosti nějakého mikroprocesoru. Tohle nebude složitý kód, takže je malá šance, že tam bude významná chyba. Může to zároveň obsahovat USB. Podobně, jako je řešen například VESC. Je to technologie, u které si myslím, že se ji do budoucna stejně nevyhneme.

**Comment by roman-dvorak** [comment]
Ano, pokud by byl tento případ, tak bych byl rozhodně pro CAN konkrétně asi UAVCAN.
